database.class.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+    include_once 'databaseManager.class.php';
+
+    /**
+     * Database connection
+     */
+    class Database extends DatabaseManager
+    {
+        /**
+         * Constructor
+         */
+        public function __construct($dbDriver, $host, $port, $username, $password, $dbName, $unixSocket, $charset)
+        {
+            parent::__construct($dbDriver, $host, $port, $username, $password, $dbName, $unixSocket, $charset);
+        }
+
+        /**
+         * @return PDO
+         */
+        public function openConnection()
+        {
+            if ($this->isConnected() != 1)
+            {
+                try 
+                {                    
+                    $this->connection = new PDO($this->dbDriver.":host=".$this->host.";port=".$this->port.";charset=".$this->charset.";unix_socket=".$this->unixSocket.";dbname=".$this->dbName,$this->username,$this->password);
+                    echo "Successfull connection to database" . "\n";
+
+                    /* disable emulated prepared statements and use real prepared statements */
+                    $this->connection->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
+                    $this->connection->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+                } catch (PDOException $e) 
+                {
+                    die($e->getMessage());
+                }  
+            }
+            return $this->connection;    
+        }
+
+        /**
+         * Returns rows from the database based on the conditions
+         * @param string name of the table
+         * @param array select, where, order_by, limit and return_type conditions
+         */
+        public function selectData($table, $conditions = array())
+        {
+            $sql = 'SELECT ';
+            $sql .= array_key_exists("select",$conditions)?$conditions['select']:'*';
+            $sql .= ' FROM '.$table;
+
+            if(array_key_exists("where",$conditions))
+            {
+                $sql .= ' WHERE ';
+                $i = 0;
+
+                foreach($conditions['where'] as $key => $value)
+                {
+                    $pre = ($i > 0)?' AND ':'';
+                    $sql .= $pre.$key." = '".$value."'";
```

Comment:
@lumakos This seems pretty vulnerable. How about handling possible SQL injection?

Reply:
Here i should espace special characters in $value using mysql_real_escape_string() in case of MySQL query.
`$sql .= $pre.$key." = '".mysql_real_escape_string($value)."'";`

Reply:
@lumakos This might be OK
BTW: the `$key` is also sql-injectable but escaping it won't do the trick, a possible approach could be to check it the column exists by checking the table metadata.

FYI: there is a better way of doing that and that's by using (prepared SQL statements)[http://php.net/manual/en/pdo.prepared-statements.php].